framework/src/base/DumpObjectsProblem.C

```diff
@@ -0,0 +1,347 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#include "DumpObjectsProblem.h"
+#include "MooseMesh.h"
+#include <sstream>
+
+#include "libmesh/string_to_enum.h"
+
+template <>
+InputParameters
+validParams<DumpObjectsProblem>()
+{
+  InputParameters params = validParams<FEProblem>();
+  params.addRequiredParam<std::string>(
```

Comment:
Could we do a vector of strings here to allow multiple action paths?

Reply:
Sure thing, I think all action expansions are build and stored anyways.

Reply:
And we could allow leaving the vector empty to output everything.